traitsui/tests/editors/test_tabular_editor.py

```diff
@@ -39,30 +45,326 @@ class Report(HasTraits):
 
     selected_row = Int(-1)
 
+    multi_selected = List(Instance(Person))
+
+    selected_rows = List(Int())
+
     # Event for triggering a UI repaint.
     refresh = Event()
 
     # Event for triggering a UI table update.
     update = Event()
 
-    traits_view = View(
-        Item(
-            name="people",
-            editor=TabularEditor(
-                adapter=ReportAdapter(),
-                selected="selected",
-                selected_row="selected_row",
-                refresh="refresh",
-                update="update",
-            ),
+
+def get_view(multi_select=False):
+    if multi_select:
+        return View(
+            Item(
+                name="people",
+                editor=TabularEditor(
+                    adapter=ReportAdapter(),
+                    selected="multi_selected",
+                    selected_row="selected_rows",
+                    refresh="refresh",
+                    update="update",
+                    multi_select=True,
+                ),
+            )
+        )
+    else:
+        return View(
+            Item(
+                name="people",
+                editor=TabularEditor(
+                    adapter=ReportAdapter(),
+                    selected="selected",
+                    selected_row="selected_row",
+                    refresh="refresh",
+                    update="update",
+                ),
+            )
+        )
+
+
+def get_selected_rows(editor):
+    """ Returns a list of all currently selected rows.
+    """
+    if is_current_backend_wx():
+        import wx
+        # "item" in this context means "row number"
+        item = -1
+        selected = []
+        while True:
+            item = editor.control.GetNextItem(
+                item, wx.LIST_NEXT_ALL, wx.LIST_STATE_SELECTED
+            )
+            if item == -1:
+                break
+            selected.append(item)
+        return selected
+
+    elif is_current_backend_qt4():
+        rows = editor.control.selectionModel().selectedRows()
+        return [r.row() for r in rows]
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def set_selected_single(editor, row):
+    """ Selects a specified row in an editor with multi_select=False.
+    """
+    if is_current_backend_wx():
+        editor.control.Select(row)
+
+    elif is_current_backend_qt4():
+        from pyface.qt.QtGui import QItemSelectionModel
+
+        smodel = editor.control.selectionModel()
+        mi = editor.model.index(row, 0)
+        smodel.select(
+            mi, QItemSelectionModel.ClearAndSelect | QItemSelectionModel.Rows
```

Comment:
If you could comment on why we need the union of these two actions that will be very helpful for future developers who try to understand these tests. It is not obvious why the `Rows` flag is needed here.